pkg/deploy/vmsscleaner/clean.go

```diff
@@ -62,3 +64,30 @@ func (c *cleaner) RemoveFailedNewScaleset(ctx context.Context, rgName, vmssToDel
 	// If it was not returned from List, assume it does not exist and that deployment can be retried.
 	return true
 }
+
+// UpdateVMSSProbes attempts to remove the probes references so we can update the load balancer rules
+func (c *cleaner) UpdateVMSSProbes(ctx context.Context, rgName string) (retry bool) {
+	scalesets, err := c.vmss.List(ctx, rgName)
+	if err != nil {
+		c.log.Warn(err)
+		return false
+	}
+
+	for _, vmss := range scalesets {
+		name := *vmss.Name
+		if !strings.HasPrefix(name, "gateway-vmss-") {
+			continue
+		}
+
+		//removes the network profile healthprobe
+		vmss.VirtualMachineProfile.NetworkProfile.HealthProbe = nil
+		c.log.Printf("removing the probe reference from the vmss %s", name)
+		err = c.vmss.CreateOrUpdateAndWait(ctx, rgName, name, vmss)
+		if err != nil {
+			c.log.Warn(err)
+			return false // If update failed, gateway vmss still exists. Don't retry.
+		}
+	}
+	// no scaleset matched, so we should not retry and return the error
+	return false
```

Comment:
here we are supposed to retry actually, this was a mistake when I first wrote that function (I think there was another commit that fixed it). Because if no scaleset is matched, we would want to recreate one, and if the update succeeded we would also want to retry